ServiceStation/src/main/java/com/servicestation/controller/CarController.java

```diff
@@ -0,0 +1,114 @@
+package com.servicestation.controller;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import com.servicestation.model.Make;
+import com.servicestation.model.Year;
+import com.servicestation.service.CarService;
+
+@Controller
+@RequestMapping("/cars")
+public class CarController {
+	@Autowired
+	private CarService carService;
+
+	@RequestMapping(value = "{clientId}", method = RequestMethod.GET)
+	public String showCars(@PathVariable("clientId") Long clientId, Model model) {
+		model.addAttribute("client", carService.getClient(clientId));
+		model.addAttribute("cars", carService.findByClientId(clientId));
+		return "cars";
+	}
+
+	@RequestMapping(value = "{clientId}/add", method = RequestMethod.GET)
+	public String showCarAddForm(@PathVariable("clientId") Long clientId, Model model) {
+		model.addAttribute("client", carService.getClient(clientId));
+		return "carform";
+	}
+
+	@RequestMapping(value = "{clientId}", method = RequestMethod.POST)
+	public String addCar(@PathVariable("clientId") Long clientId,
+			@RequestParam(value = "yearId") Long yearId,
+			@RequestParam(value = "makeId") Long makeId,
+			@RequestParam(value = "modelId") Long modelId, 
+			@RequestParam(value = "vin") String vin, Model model) {
+		carService.save(clientId, yearId, makeId, modelId,vin);
+		model.addAttribute("clientId", clientId);
+		return "redirect:/cars/{clientId}";
+	}
+
+	@RequestMapping(value = "{clientId}/{carId}/delete")
+	public String deleteCar(@PathVariable("clientId") Long clientId, @PathVariable("carId") Long carId, Model model) {
```

Comment:
@jan843 Could use DELETE method and @DeleteMapping
Also could you review {clientId}/{carId}/ REST path?